packages/cli/src/oclif/commands/migrate.js

```diff
@@ -6,6 +7,49 @@ const { callAPI } = require('../../utils/api');
 const { buildFlags } = require('../buildFlags');
 
 class MigrateCommand extends BaseCommand {
+  async run_require_confirmation_pre_checks(app, requestBody) {
+    const assumeYes = 'yes' in this.flags;
+    const url = `/apps/${app.id}/pre-migration-require-confirmation-checks`;
+
+    this.startSpinner(`Running pre-checks before migration...`);
+
+    try {
+      await callAPI(
+        url,
+        {
+          method: 'POST',
+          body: requestBody,
+        },
+        true,
+      );
+    } catch (response) {
+      this.stopSpinner();
+
+      // 409 from the backend specifically signals pre-checks failed
+      if (response.status === 409) {
+        const softCheckErrors = _.get(response, 'json.errors');
+        const formattedErrors = softCheckErrors.map((e) => `* ${e}`).join('\n');
```

Comment:
hmmm if `response.json` or `response.json.errors` is undefined, `.map()` will throw an error right? would it be viable to do something like this so we don't get runtime exceptions? 

```
const softCheckErrors = _.get(response, 'json.errors', []);
```

Reply:
Good catch! Will add